test/Operators/hybrid/cuda.jl

```diff
@@ -5,35 +5,129 @@ import ClimaCore:
+
+    horzdomain = Domains.RectangleDomain(
+        Geometry.XPoint{FT}(xlim[1]) .. Geometry.XPoint{FT}(xlim[2]),
+        Geometry.YPoint{FT}(ylim[1]) .. Geometry.YPoint{FT}(ylim[2]),
+        x1periodic = true,
+        x2periodic = true,
+    )
+    horzmesh = Meshes.RectilinearMesh(horzdomain, xelem, yelem)
+
+    quad = Spaces.Quadratures.GLL{npoly + 1}()
+
+    # Define horz topology and space
+    horztopology = Topologies.Topology2D(context, horzmesh)
+    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
+
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
+    return (hv_center_space, hv_face_space)
+end
+
+function hvspace_3D_sphere(context)
+    FT = Float64
+
+    # Define vert domain and mesh
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(0.0),
+        Geometry.ZPoint{FT}(1.0);
+        boundary_tags = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = 10)
+
+    # Define vert topology and space
+    verttopology = Topologies.IntervalTopology(context, vertmesh)
+    vert_center_space = Spaces.CenterFiniteDifferenceSpace(verttopology)
+
+    # Define horz domain and mesh
+    horzdomain = Domains.SphereDomain(FT(30.0))
+    horzmesh = Meshes.EquiangularCubedSphere(horzdomain, 4)
+
+    quad = Spaces.Quadratures.GLL{3 + 1}()
+
+    # Define horz topology and space
+    horztopology = Topologies.Topology2D(context, horzmesh)
+    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
+
+    # Define hv spaces
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
+    return hv_center_space, hv_face_space
+end
+
+@testset "Finite difference GradientF2C CUDA" begin
+    gpu_context = ClimaComms.SingletonCommsContext(ClimaComms.CUDADevice())
+    device = ClimaComms.device() #ClimaComms.CUDADevice()
+    println("running test on $device device")
+
+    # Define hv GPU space
+    hv_center_space_gpu, hv_face_space_gpu = hvspace_3D_sphere(gpu_context)
+
+    z = Fields.coordinate_field(hv_face_space_gpu).z
+
+    gradc = Operators.GradientF2C()
+
+    @test parent(Geometry.WVector.(gradc.(z))) ≈
+          parent(Geometry.WVector.(ones(hv_center_space_gpu)))
+end
+
+@testset "2D SE, 1D FD Extruded Domain ∇ ODE Solve horizontal CUDA" begin
+
+    # Advection Equation
+    # ∂_t f + c ∂_x f  = 0
+    # the solution translates to the right at speed c,
+    # so if you you have a periodic domain of size [-π, π]
+    # at time t, the solution is f(x - c * t, y)
+    # here c == 1, integrate t == 2π or one full period
+
+    function rhs!(dudt, u, _, t)
+        # horizontal divergence operator applied to all levels
+        hdiv = Operators.Divergence()
+        @. dudt = -hdiv(u * Geometry.UVVector(1.0, 1.0))
```

Comment:
@simonbyrne , @charleskawczynski ,
I'm getting an error on this line: 
```
ERROR: GPU compilation of MethodInstance for ClimaCore.Operators.copyto_spectral_kernel!
```
but the same [test](https://github.com/CliMA/ClimaCore.jl/blob/2045467aa8343d68b99ac8e3cf9f34bc635c46ff/test/Operators/hybrid/3d.jl#L153) passes on CPU. Can you please advise?

Reply:
What's the full error + stacktrace?

Reply:
Resolved in commit [c7822d6](https://github.com/CliMA/ClimaCore.jl/pull/1232/commits/c7822d67785a433c06702dc6c1de13f594f0227f)!